### PR TITLE
Canvas.py doc enhancement

### PR DESCRIPTION
**Pull request by Nusnus**

### CODE REVIEW

File: celery/canvas.py [hunk 1]
```diff
@@ -1448,6 +1448,14 @@ class group(Signature):
 
     @classmethod
     def from_dict(cls, d, app=None):
+        """Create a group signature from a dictionary of tasks.
```
Comment (thedrow) [inline celery/canvas.py:1451]: This is not a dictionary of tasks but a dictionary which represents a group signature. Within it, there is a list of tasks in the group.
Please find a way to rephrase.

File: celery/canvas.py [hunk 2]
```diff
@@ -1684,15 +1750,16 @@ def _freeze_group_tasks(self, _id=None, group_id=None, chord=None,
...
-            # tasks1, tasks2 are each a clone of self.tasks
+            # When the group tasks are a generator, we need to make sure we don't
+            # exhaust it during the freeze process. We use two generators to do this.
+            # One generator will be used to freeze the tasks to get their AsyncResult.
+            # The second generator will be used to replace the tasks in the group with an unexhausted state.
+
+            # Create two new generators from the original generator of the group tasks (cloning the tasks).
             tasks1, tasks2 = itertools.tee(self._unroll_tasks(self.tasks))
-            # freeze each task in tasks1, results now holds AsyncResult for each task
+            # Use the first generator to freeze the group tasks to aquire the AsyncResult for each task.
```
Comment (thedrow) [inline celery/canvas.py:1760]:
```suggestion
            # Use the first generator to freeze the group tasks to acquire the AsyncResult for each task.
```

**Review by thedrow — CHANGES_REQUESTED** [review]
There are a few minor issues to address.
Overall, this is good.

File: celery/canvas.py (hunk 1 above)
Comment (Nusnus) [reply celery/canvas.py]: Fix pushed.

**Review by thedrow — APPROVED** [review]

**Pull request merged**